phoenix-core/src/it/java/org/apache/phoenix/end2end/index/IndexRegionObserverMutationBlockingIT.java

```diff
@@ -0,0 +1,292 @@
+                    "standby-zk-url", 
+                    ClusterRoleRecord.ClusterRole.STANDBY, 
+                    1L);
+            haAdmin.createOrUpdateDataOnZookeeper(blockingCrr);
+            
+            Thread.sleep(ZK_CURATOR_EVENT_PROPAGATION_TIMEOUT_MS);
+            
+            // Test that UPSERT with local index also throws MutationBlockedIOException
+            try {
+                conn.createStatement().execute("UPSERT INTO " + dataTableName + 
+                    " VALUES ('1', 'Alice', 28)");
+                conn.commit();
+                fail("Expected MutationBlockedIOException to be thrown for local index");
+            } catch (CommitException e) {
+                assertTrue("Expected mutation blocked exception", 
+                    containsMutationBlockedException(e));
+            }
+        }
+    }
+
+    @Test
+    public void testMutationAllowedWhenNotBlocked() throws Exception {
+        String dataTableName = generateUniqueName();
+        String indexName = generateUniqueName();
+        
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            // Create data table and index
+            conn.createStatement().execute("CREATE TABLE " + dataTableName + 
+                " (id VARCHAR PRIMARY KEY, name VARCHAR, age INTEGER)");
+            conn.createStatement().execute("CREATE INDEX " + indexName + 
+                " ON " + dataTableName + "(name)");
+            
+            // Set up CRR in ACTIVE state (should not block)
+            ClusterRoleRecord nonBlockingCrr = new ClusterRoleRecord("active_test",
+                    HighAvailabilityPolicy.FAILOVER, 
+                    haAdmin.getZkUrl(), 
+                    ClusterRoleRecord.ClusterRole.ACTIVE,
+                    "standby-zk-url", 
+                    ClusterRoleRecord.ClusterRole.STANDBY, 
+                    1L);
+            haAdmin.createOrUpdateDataOnZookeeper(nonBlockingCrr);
+            
+            Thread.sleep(ZK_CURATOR_EVENT_PROPAGATION_TIMEOUT_MS);
+            
+            // Mutations should work fine
+            conn.createStatement().execute("UPSERT INTO " + dataTableName + 
+                " VALUES ('1', 'Bob', 35)");
+            conn.createStatement().execute("UPSERT INTO " + dataTableName + 
+                " VALUES ('2', 'Carol', 27)");
+            conn.commit();
+            
+            // Verify data was inserted successfully
+            try (java.sql.ResultSet rs = conn.createStatement().executeQuery(
+                "SELECT COUNT(*) FROM " + dataTableName)) {
+                assertTrue(rs.next());
+                assertEquals(2, rs.getInt(1));
+            }
+        }
+    }
+
+    @Test
+    public void testMutationBlockingTransition() throws Exception {
+        String dataTableName = generateUniqueName();
+        String indexName = generateUniqueName();
+        
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
```

Comment:
Why are we creating connection with a specific URL? Shouldn't we use some generic getConnection method that will lookup in CRR which cluster is active and create connection to that cluster?